### ISSUE TITLE
A way to use `Molecule.from_file` with an explicitly-protonated .sdf?

### ISSUE THREAD

**Issue by zanebeckwith** [openforcefield/openff-toolkit#1871]
If I have a molecule in .sdf format that I've explicitly protonated, what's the recommended way to read that into a `Molecule` such that I keep that protonation? (Context: I'm using the RDKit toolkit)

It looks like `Molecule.from_file` ends up calling RDKit's `AddHs`, and the `hydrogens_are_explicit` flag isn't exposed for `from_file`. 

I could read the .sdf into an RDKit Mol myself and then call `Molecule.from_rdkit` (specifying `hydrogens_are_explicit=True`), but it looks like `from_file` does quite a bit of additional sanitizing, which would be lost in that case.

Apologies if I'm misunderstanding something! I'm not an RDKit power user.

Thanks for the excellent project!

**Comment by mattwthompson** [comment]
`Molecule.from_file` is the recommended way to load SDF files, see for example loading a zwitterion (which keeps hydrogens and protonation state): https://docs.openforcefield.org/projects/toolkit/en/stable/users/molecule_cookbook.html#from-sdf-file


```python
In [8]: [
   ...:     atom.formal_charge.m
   ...:     for atom in Molecule(
   ...:         "/Users/mattthompson/software/openff-toolkit/docs/users/zw_l_alanine.sdf"
   ...:     ).atoms
   ...: ]
Out[8]: [0, 0, 1, 0, -1, 0, 0, 0, 0, 0, 0, 0, 0]
```

If this isn't working for your use case, please share a [reproduction](https://stackoverflow.com/help/minimal-reproducible-example) so we can dig into it deeper

**Comment by j-wags** [comment]
Yeah, just to echo Matt here - My intention with `Molecule.from_file` is to preserve protonation state/formal charge exactly. The `AddHs` calls can be a little counterintuitive/they can behave differently in different contexts. So please do share your input file if you can so we can dig into your case :-)

**Comment by zanebeckwith** [comment]
> My intention with Molecule.from_file is to preserve protonation state/formal charge exactly

Ah, OK, it sounds like I have nothing to worry about, then!

I haven't actually seen any undesired behavior. I just noticed the `hydrogens_are_explicit` flag in the other functions, which gates the call to `AddHs`, and assumed that without it `from_file` wouldn't work if I already protonated.

> The AddHs calls can be a little counterintuitive

Yeah, interesting, I just assumed that `AddHs` would steamroll any preexisting protonation.

I'm sorry for the noise! I just didn't do my homework. Thanks for the help @mattwthompson and @j-wags
